File: src/main/java/com/huangyueran/spark/operator/Intersection.java
package com.huangyueran.spark.operator;

import com.huangyueran.spark.utils.SparkUtils;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

/**
 * @category 返回两个RDD的交集--Returns the intersection of two RDD 
 * @author huangyueran
 * @time 2019-7-21 16:38:20
 */
public class Intersection {

	public static void main(String[] args) {
		JavaSparkContext sc = SparkUtils.getLocalSparkContext(Intersection.class);

		intersection(sc);
	}

	static void intersection(JavaSparkContext sc) {
		List<String> datas1 = Arrays.asList("张三", "李四", "tom");
		List<String> datas2 = Arrays.asList("tom", "gim");

		/**
		 *  =====================================
		 *   |             返回两个RDD的交集                                                   |
		 *   |             Returns the intersection of two RDD                    |                                                                                                                                                                                                                                    | 
		 *   =====================================
		 */
		JavaRDD<String> intersectionRDD = sc.parallelize(datas1).intersection(sc.parallelize(datas2));

		intersectionRDD.foreach(new VoidFunction<String>() {

			@Override
			public void call(String t) throws Exception {
				System.out.println(t);
			}
		});

	}

}

File: src/main/scala/com/hyr/spark/operator/Intersection.scala
package com.hyr.spark.operator

import org.apache.spark.{SparkConf, SparkContext}

/** *****************************************************************************
  * @date 2019-08-08 10:32
  * @author: <a href=mailto:huangyr>黄跃然</a>
  * @Description: 返回两个RDD的交集--Returns the intersection of two RDD
  ******************************************************************************/
object Intersection {

  def intersection(sparkContext: SparkContext): Unit = {
    val data1 = List("张三", "李四", "tom")
    val data2 = List("tom", "gim")
    val rdd = sparkContext.makeRDD(data1).intersection(sparkContext.makeRDD(data2))
    rdd.foreach(t => {
      println(t)
    })
  }

  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf setAppName "Intersection" setMaster "local"
    val sparkContext = new SparkContext(sparkConf)

    intersection(sparkContext)
  }


}
